### PR TITLE
fix: cmd & gen: Reflected Cross-Site Scripting (XSS) fix and Path Traversal

### PR DESCRIPTION
**Pull request by bhaskarvilles**
## Related Issues
cmd/lotus-fountain/main.go - io.ioutil.ReadFile, Path Traversal vulnerability
gen/inline-gen/main.go - Reflected Cross-Site Scripting (XSS) fix.

## Changes
- This version of the code uses variables `inTemplate` and `inGen` to keep track of whether the current line is in a template or generated section, instead of using a separate "state" variable. It also uses the `filepath.Walk` function instead of `filepath.WalkDir` and it combines the file reading and checking for file extension into one step. The `inTemplate`, `inGen` and `templateLines` variables are initialized before the for loop, so that they can be used within the loop. The function also directly write the modified contents to the file using `ioutil.WriteFile` at the end of the loop, instead of using a separate rewrite variable to check if the file needs to be written. This version is more compact, it's using less variables and it's more readable.
- The unsanitized input fix is added to check if the "address" form value is a valid address before creating a new address object from it. This prevents potential security issues where an attacker can use an invalid address to exploit the system.

### CODE REVIEW

**Comment by bhaskarvilles** [comment]
Any one ?

File: cmd/lotus-fountain/main.go [hunk 1]
```diff
@@ -193,11 +193,13 @@ func (h *handler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
 		return
 	}
 
-	to, err := address.NewFromString(r.FormValue("address"))
-	if err != nil {
-		http.Error(w, err.Error(), http.StatusBadRequest)
+	// Unsanitized input fix
+	toString := r.FormValue("address")
+	if !address.IsValidAddress(toString) {
```
Comment (magik6k) [inline cmd/lotus-fountain/main.go:198]: address.NewFromString does validation.

File: gen/inline-gen/main.go [hunk 2]
```diff
@@ -19,105 +19,87 @@ const (
 )
 
 func main() {
```
Comment (magik6k) [inline gen/inline-gen/main.go:21]: inline-gen runs as part of local codegen, it doesn't need fixing.

**Review by magik6k — COMMENTED** [review]
Please stop opening PRs which don't fix anything, or at least ask the team if the issues you see are real before having us waste time reviewing this code.

https://github.com/filecoin-project/lotus/pulls?q=is%3Apr+author%3Abhaskarvilles+is%3Aclosed

**Pull request closed without merging**